azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageRampupServlet.java

```diff
@@ -73,38 +78,45 @@ public void init(final ServletConfig config) throws ServletException {
   }
 
   @Override
-  protected void handleGet(HttpServletRequest req, HttpServletResponse resp, Session session)
+  protected void handleGet(
+      final HttpServletRequest req, final HttpServletResponse resp, final Session session)
       throws ServletException, IOException {
     /* Get specific record */
     try {
       String response = null;
       if (BASE_IMAGE_RAMPUP_URI.equals(req.getRequestURI())) {
         // imageType must present. If not present throws ImageMgmtValidationException
-        String imageType = HttpRequestUtils.getParam(req, IMAGE_TYPE);
+        final String imageType = HttpRequestUtils.getParam(req, IMAGE_TYPE);
         if (imageType == null) {
           log.error("Image type can't be null. Must provide valid image type to get active rampup"
               + " plan");
-          throw new ImageMgmtValidationException(
+          throw new ImageMgmtValidationException(ErrorCode.BAD_REQUEST,
               "Image type can't be null. Must provide valid image type to get rampup plan.");
         }
+        // Check for required permission to invoke the API
+        if (!hasImageManagementPermission(imageType, session.getUser(), Type.GET)) {
```

Comment:
How is the READ_ONLY_PERMISSION used? Here we're directly using `Type.GET`

Reply:
Now GUEST_PERMISSION only can have only get or read access.